### ISSUE TITLE
Question about weight decay optimizer on TPU

### ISSUE THREAD

**Issue by ymcui** [zihangdai/xlnet#66]
Hello,

I've been experimenting XLNet on TPU.
One quick question about the weight decay optimizer in `model_utils.py`.
https://github.com/zihangdai/xlnet/blob/master/model_utils.py#L125

I saw there are two options here, depending on the value of `FLAGS.weight_decay`.
- if no weight decay, it will go with `AdamOptimizer`
- if with weight decay (say 0.01), it should go with `AdamWeightDecayOptimizer`. 

However, there is another condition that `FLAGS.num_core_per_host == 1`, which is not friendly for TPU, as `num_core_per_host` usually >=8 for TPUs. 
So if `weight_decay > 0` and using TPU (`num_core_per_host >1`) it will raise `ValueError`.

In your paper, according to the appendix, you set `weight_decay=0.01` and pre-trained XLNet using TPU, and it should have raised `ValError` if using the code in this repository.

I am not sure if this a bug or an intended action.
Thank you for your response.

**Comment by kimiyoung** [comment]
Good catch. It's a bug. Will fix it soon.
